src/state/user/hooks.tsx

```diff
@@ -105,19 +105,21 @@ export function useExpertModeManager(): [boolean, () => void] {
   return [expertMode, toggleSetExpertMode]
 }
 
-export function useUserSingleHopOnly(): [boolean, (newSingleHopOnly: boolean) => void] {
+export function useIsLegacyRouter(): [boolean, (userLegacyRouter: boolean) => void] {
+  const { chainId } = useActiveWeb3React()
+
   const dispatch = useAppDispatch()
 
-  const singleHopOnly = useAppSelector((state) => state.user.userSingleHopOnly)
+  const legacyRouter = useAppSelector((state) => state.user.userLegacyRouter)
 
-  const setSingleHopOnly = useCallback(
-    (newSingleHopOnly: boolean) => {
-      dispatch(updateUserSingleHopOnly({ userSingleHopOnly: newSingleHopOnly }))
+  const setLegacyRouter = useCallback(
+    (newLegacyRouter: boolean) => {
+      dispatch(updateUserLegacyRouter({ userLegacyRouter: newLegacyRouter }))
     },
     [dispatch]
   )
 
-  return [singleHopOnly, setSingleHopOnly]
+  return [legacyRouter || chainId !== SupportedChainId.MAINNET, setLegacyRouter]
```

Comment:
this is a bit messy because `setLegacyRouter(!legacyRouter)` won't do anything. instead you want to show the toggle as disabled when not on mainnet

Reply:
good call, design preferred hiding the toggle entirely on non mainnet.